bitcoin/tx.c

```diff
@@ -700,6 +700,7 @@ struct bitcoin_tx *fromwire_bitcoin_tx(const tal_t *ctx,
 {
 	struct bitcoin_tx *tx;
 
+	fromwire_u32(cursor, max);
```

Comment:
is there a way to constrain `pull_bitcoin_tx` to use the length we just deserialized?  or at least check that those many bytes were consumed after the fact?

Reply:
Added a post-parse check

Reply:
We could also replace the `max` pointer with a pointer to the read prefix length, this way the tx parsing code itself will abort if a read would go beyond the supposed end of the tx stream. Would still need to have a check that the length was consumed completely (`max` gets decremented at each read and if it matches we should end up with a `max` of 0).